### PR TITLE
Add -XX cmdline option for TransparentHugepage

### PR DESCRIPTION
**Pull request by fengxue-IS**
Related: #3611 

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

File: runtime/oti/jvminit.h [hunk 1]
```diff
@@ -451,6 +451,9 @@ enum INIT_STAGE {
 #define VMOPT_XCOMPRESSEDREFS "-Xcompressedrefs"
 #define VMOPT_XNOCOMPRESSEDREFS "-Xnocompressedrefs"
 
+#define VMOPT_XENABLE_TRANSPARENT_HUGEPAGE "-XenableTransparentHugepage"
+#define VMOPT_XDISABLE_TRANSPARENT_HUGEPAGE "-XdisableTransparentHugepage"
```
Comment (DanHeidinga) [inline runtime/oti/jvminit.h]: Please use an `-XX:[+-]TransparentHugePage` option.  XX options are preferable to X ones

Comment (fengxue-IS) [reply runtime/oti/jvminit.h]: Updated

**Comment by pshipton** [comment]
Don't forget to open a docs issue, or update https://github.com/eclipse/openj9-docs/issues/273 to include the option.

**Comment by fengxue-IS** [comment]
@pshipton will update the doc issue, thanks for the reminder

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -1815,6 +1815,18 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
...
+			/* -XX commandline option for +/- TransparentHugepage */
+			argIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOTRANSPARENT_HUGEPAGE, NULL);
+			argIndex2 = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXTRANSPARENT_HUGEPAGE, NULL);
+			{
+				/* last instance of +/- TransparentHugepage found on the command line wins */
+				if (argIndex2 > argIndex) {
+					j9port_control(J9PORT_CTLDATA_VMEM_ADVISE_HUGEPAGE, 1);
+				} else if (argIndex > argIndex2) {
+					j9port_control(J9PORT_CTLDATA_VMEM_ADVISE_HUGEPAGE, 0);
+				}
```
Comment (DanHeidinga) [inline runtime/vm/jvminit.c]: This defaults to not madvising transparent huge pages.  @vijaysun-omr Do you have data indicating which way this should default?

Comment (DanHeidinga) [reply runtime/vm/jvminit.c]: ping @vijaysun-omr.  Do you have data on indicating which way madavise for transparent largepages should default?  I suspect the default should be "enabled"

**Comment by vijaysun-omr** [comment]
I believe we had seen a performance disadvantage vs Hotspot on one of our SpringBoot throughput scenarios with the transparent huge page setting at "madvise" on the machine where the testing was being done and OpenJ9 performance improved when we forced it to use transparent huge pages. I feel we should therefore be using transparent huge pages when the setting is at "madvise".

File: runtime/vm/jvminit.c [hunk 3]
```diff
@@ -1815,6 +1815,18 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 				}
 			}
 
+			/* -XX commandline option for +/- TransparentHugepage */
+			argIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOTRANSPARENT_HUGEPAGE, NULL);
+			argIndex2 = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXTRANSPARENT_HUGEPAGE, NULL);
+			{
+				/* last instance of +/- TransparentHugepage found on the command line wins */
+				if (argIndex2 > argIndex) {
```
Comment (DanHeidinga) [inline runtime/vm/jvminit.c]:
```suggestion
				if (argIndex2 >= argIndex) {
```

**Comment by dmitripivkine** [comment]
I believe this option should be on by default (always madvise)

File: runtime/vm/jvminit.c [hunk 4]
```diff
@@ -1815,6 +1815,18 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 				}
 			}
 
+			/* -XX commandline option for +/- TransparentHugepage */
+			argIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOTRANSPARENT_HUGEPAGE, NULL);
+			argIndex2 = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXTRANSPARENT_HUGEPAGE, NULL);
+			{
+				/* last instance of +/- TransparentHugepage found on the command line wins */
```
Comment (DanHeidinga) [inline runtime/vm/jvminit.c]:
```suggestion
				/* last instance of +/- TransparentHugepage found on the command line wins
				 * Default to -XX:+TransparentHugepage for performance reasons
				 */
```

**Comment by fengxue-IS** [comment]
Update code to set `-XX:+TransparentHugepage` by default

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,zlinux jdk8

**Pull request merged**